compass/ocean/mesh/lts_regions.py

```diff
@@ -0,0 +1,310 @@
+import math
+import subprocess as sp
+
+import netCDF4 as nc
+import numpy as np
+import xarray
+from shapely.geometry import Point
+from shapely.geometry.polygon import Polygon
+
+from compass.step import Step
+
+
+class LTSRegionsStep(Step):
+    """
+    A step for adding LTS regions to a global MPAS-Ocean mesh
+
+    Attributes
+    ----------
+    cull_mesh_step : CullMeshStep
+        The culled mesh step containing input files to this step
+    """
+    def __init__(self, test_case, cull_mesh_step,
+                 name='lts_regions', subdir='cull_mesh'):
+        """
+        Create a new step
+
+        Parameters
+        ----------
+        test_case : compass.ocean.tests.global_ocean.mesh.Mesh
+            The test case this step belongs to
+
+        cull_mesh_step : cull_mesh_step
+            The culled mesh step containing input files to this step
+
+        name : str, optional
+            the name of the step
+
+        subdir : str, optional
+            the subdirectory for the step.  The default is ``name``
+        """
+        super().__init__(test_case, name=name, subdir=subdir,
+                         cpus_per_task=None, min_cpus_per_task=None)
+        self.cull_mesh_step = cull_mesh_step
+
+    def setup(self):
+        """
+        Set up the test case in the work directory, including downloading any
+        dependencies.
+        """
+        super().setup()
+
+        config = self.cull_mesh_step.config
+        self.cpus_per_task = config.getint('spherical_mesh',
+                                           'cull_mesh_cpus_per_task')
+        self.min_cpus_per_task = config.getint('spherical_mesh',
+                                               'cull_mesh_min_cpus_per_task')
+
+        self.lts_fine_regions_file = 'hurricane_lts_fine_regions.nc'
+
+        self.add_input_file(
+            filename=self.lts_fine_regions_file,
+            target='hurricane_lts_fine_regions.nc',
+            database='hurricane')
+
+    def run(self):
+        """
+        Run this step of the test case
+        """
+        file_name = 'hurricane_lts_fine_regions.nc'
+        fine_rgn = nc.Dataset(file_name, 'r', format='NETCDF4_64BIT')
+
+        region = 'east_us_coast'
```

Comment:
Presumably, this should be an input (e.g. a config option) instead of being hard-coded?

Reply:
Eventually we will have an automated framework to set the LTS regions so we would not need to read them from an input file anymore. For the time being, if that's OK, we should probably keep it specific to hurricane.